chains/celo/listener_test.go

```diff
@@ -30,3 +66,46 @@ func TestListener_start_stop(t *testing.T) {
 	// Initiate shutdown
 	l.close()
 }
+
+// Testing transaction Block hash
+func TestListener_TransactionBlockHash(t *testing.T) {
+
+	l := createTestListener(t)
+	err := l.start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create and submit a new transaction and return the signed transaction hash
+	hash := newTransaction(t, l)
+	// get the block hash of the receipt
+	blockHash := l.getTransactionBlockHash(hash)
+	// using the receipt blockhash get the block
+	block, err := l.conn.Client().BlockByHash(context.Background(), blockHash)
```

Comment:
You should be querying the receipt of the tx here and verifying the block from that